### PR TITLE
Add Mechanism2d article

### PR DESCRIPTION
**Pull request by Starlight220**
Waiting on wpilibsuite/allwpilib#3281.

RLIs are broken until that PR is merged and perhaps released (if we don't want RLIs to point to specific untagged commits).

### CODE REVIEW

**Comment by Daltz333** [comment]
Untagged commits are fine

File: source/docs/software/wpilib-tools/glass/mech2d-widget.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
+The Mechanism2d Widget
+======================
+
+Glass supports displaying stick-figure representations of your robot's mechanisms using the :guilabel:`Mechanism2d` widget. An instance of the ``Mechanism2d`` class should be created and populated, sent over NetworkTables, and updated periodically with the latest mechanism states in your robot code.
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/glass/mech2d-widget.rst:4]: What usefulness does showing stick figure representatives of mechanisms showcase? What mechanisms are supported? I'd like to expand this intro paragraph a bit.

File: source/docs/software/wpilib-tools/glass/mech2d-widget.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+Creating and Configuring the Mechanism2d Instance
+-------------------------------------------------
+
+The ``Mechanism2d`` object is the base of the mechanism. To get a root node (represented by a ``MechanismRoot2d`` object), call ``getRoot(name, x, y)`` on the container ``Mechanism2d`` object. Coordinate system follow the same orientation as Field2d - ``(0,0)`` is bottom left.
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/glass/mech2d-widget.rst:9]:
```suggestion
The ``Mechanism2d`` object is the base of the mechanism. To get a root node (represented by a ``MechanismRoot2d`` object), call ``getRoot(name, x, y)`` on the container ``Mechanism2d`` object. The coordinate system follow the same orientation as Field2d - ``(0,0)`` is bottom left.
```

File: source/docs/software/wpilib-tools/glass/mech2d-widget.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  .. group-tab:: C++
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/ff52f207ccb685748fb71e1c1726ee98130e4c81/wpilibcExamples/src/main/cpp/examples/Mechanism2d/cpp/Robot.cpp
+        :language: cpp
+        :lines: 37-43
+        :linenos:
+        :lineno-start: 37
+
+Viewing the Mechanism2d widget in Glass is very similar to :ref:`viewing the Field2d widget <docs/software/wpilib-tools/glass/field2d-widget:Viewing the Robot Pose in Glass>`.
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/glass/mech2d-widget.rst:92]:
```suggestion
Viewing the Mechanism2d widget in Glass is very similar to :ref:`viewing the Field2d widget <docs/software/wpilib-tools/glass/field2d-widget:Viewing the Robot Pose in Glass>`.
```

I'd still like a picture of this in here. It's fine to duplicate a sentence here from the Glass docs.

**Review by Daltz333 — CHANGES_REQUESTED** [review]

**Comment by Daltz333** [comment]
Any update on this or stale?

**Comment by Starlight220** [comment]
I don't like to say this, but stale.

**Comment by sciencewhiz** [comment]
This will need RLIs updated for wpilibsuite/allwpilib#3589, but other then that I think it's ready for review again.

**Comment by sciencewhiz** [comment]
RLI's updated, ready for review

File: source/docs/software/dashboards/glass/mech2d-widget.rst [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+The ``Mechanism2d`` object is the "canvas" where the mechanism is drawn. The root node is where the mechanism is anchored to ``Mechanism2d``. For a single jointed arm this would the pivot point. For an elevator, this would be where it's attached to the robot's base. To get a root node (represented by a ``MechanismRoot2d`` object), call ``getRoot(name, x, y)`` on the container ``Mechanism2d`` object. The coordinate system follows the same orientation as Field2d - ``(0,0)`` is bottom left.
+
+In the examples below, an elevator is drawn, with a rotational wrist on top of the elevator. The full mechanism2d example is available in `Java <https://github.com/wpilibsuite/allwpilib/blob/main/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java>`__ / `C++ <https://github.com/wpilibsuite/allwpilib/blob/main/wpilibcExamples/src/main/cpp/examples/Mechanism2d/cpp/Robot.cpp>`__
+
+.. tabs::
+
+  .. group-tab:: Java
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/a446c25598546813ff410654bea343b2d7bc0e6c/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java
```
Comment (Daltz333) [inline source/docs/software/dashboards/glass/mech2d-widget.rst:17]:
```suggestion
     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/a446c25598546813ff410654bea343b2d7bc0e6c/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java
```

What is "climber"? Can this be any name?

File: source/docs/software/dashboards/glass/mech2d-widget.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        :lineno-start: 59
+
+
+Each ``MechanismLigament2d`` object represents a stage of the mechanism. Call ``append()``/``Append()`` on a root node or ligament node to add another node to the figure. In Java, pass a constructed ``MechanismLigament2d`` object to add it. In C++, pass the construction parameters in order to construct and add a ligament.
+
+.. tabs::
+
+  .. group-tab:: Java
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/a446c25598546813ff410654bea343b2d7bc0e6c/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java
```
Comment (Daltz333) [inline source/docs/software/dashboards/glass/mech2d-widget.rst:38]: What is `kElevatorMinimumLength`? What units? I see `90` being passed as an argument, what about mechanisms with no angles?

**Review by Daltz333 — CHANGES_REQUESTED** [review]
I'd like to see a bit more of an example on integrating the physics into mechanism2d.

**Comment by sciencewhiz** [comment]
> I'd like to see a bit more of an example on integrating the physics into mechanism2d.

I'm not sure that's appropriate for this article, which is in the glass section. That seems more appropriate for expansion in simulation.

**Comment by Daltz333** [comment]
That is fair, but this article alone really doesn't add much value for a user. I can merge it as is, but I'd really like to see a bit more expansion.

File: source/docs/software/dashboards/glass/mech2d-widget.rst (hunk 4 above)
Comment (sciencewhiz) [reply source/docs/software/dashboards/glass/mech2d-widget.rst:17]: added description of the name

File: source/docs/software/dashboards/glass/mech2d-widget.rst (hunk 5 above)
Comment (sciencewhiz) [reply source/docs/software/dashboards/glass/mech2d-widget.rst:38]: added description of parameters

**Review by Starlight220 — COMMENTED** [review]
Thanks for picking up this PR!
Just make sure to correctly capitalize `NetworkTable`(s), `Mechanism2d`, etc., and code-format stuff like 
> The `x / y` coordinate system ... `Field2d`

---
Additionally, the coordinate system is talked about twice. I'd cut down one of them.

**Comment by sciencewhiz** [comment]
@Starlight220 one place is discussing the x/y coordinates where they're used when creating Mechanism2d, and the other is discussing the parameters for the ligaments, so they aren't really the same thing. Or do you see another place where there's duplication.

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**